### PR TITLE
only do docker builds/intregration tests on changes to deps/src

### LINKED ISSUE #12161: only run integration tests / docker builds CI on code changes
### First check

- [X] I am a contributor to the Prefect codebase

### Description

something as innocuous as changing the release notes should not cause us to run 20+ multi-minute jobs that are primarily there to validate code changes

### Impact

wasting CI cycles

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zzstoatzz**
closes #12161

I don't see why we'd need to build images / do integration tests for changes outside of any of these paths, but let me know if there's some reason I'm missing

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -5,13 +5,23 @@ on:
       - .github/workflows/integration-tests.yaml
       - "**/*.py"
       - requirements.txt
+      - requirements-client.txt
```
Comment (chrisguidry) [inline .github/workflows/integration-tests.yaml:8]: :bow:

File: .github/workflows/docker-images.yaml [hunk 2]
```diff
@@ -7,9 +7,19 @@ on:
   # On each commit merged into main, push sha and branch patterns to prefect-dev
   push:
     branches: main
+    paths:
+      - 'Dockerfile'
```
Comment (chrisguidry) [inline .github/workflows/docker-images.yaml:11]: Changing `.dockerignore` could also alter the contents of the image, worth adding that one

Comment (chrisguidry) [reply .github/workflows/docker-images.yaml:11]: Oh we also need the `./ui` path: https://github.com/PrefectHQ/prefect/blob/main/Dockerfile#L31

Comment (chrisguidry) [reply .github/workflows/docker-images.yaml:11]: A few more that rarely change but that could technically influence how the package is built:

* MANIFEST.in
* setup.cfg
* versioneer.py
* .gitignore
* .gitattributes

**Review by chrisguidry — APPROVED** [review]
That's most of what I could think of, great call to do this!

**Review by desertaxle — APPROVED** [review]

File: .github/workflows/docker-images.yaml (hunk 2 above)
Comment (zzstoatzz) [reply .github/workflows/docker-images.yaml:11]: great calls! updated in [cc7989c](https://github.com/PrefectHQ/prefect/pull/12163/commits/cc7989c408fc0ae1fc744a845e8fdc6f6b952f27)

**Pull request merged**